samples/boards/nrf52/mesh/onoff_level_lighting_vnd_app/src/main.c

```diff
@@ -203,4 +209,17 @@ void main(void)
 
 	short_time_multireset_bt_mesh_unprovisioning();
 	k_timer_start(&reset_counter_timer, K_MSEC(7000), 0);
+
+#ifdef ENABLE_SMP_SVR
+	/* Initialize the Bluetooth mcumgr transport. */
+	smp_bt_register();
+
+	while (1) {
+		/* The system work queue handles all incoming mcumgr requests.
```

Comment:
Is this needed? I don't know how mcumgr works in detail but I would imagine that there might be away to repeatedly queue work to do at timer expiration from the other file as well. Or does the system work queue not have enough stack and so you're putting this on the main thread or something?

Reply:
I am too unaware about background implementation of mcumgr & have simply followed the coding style mentioned in samples/subsys/mgmt/mcumgr/smp_svr.

Reply:
I've replaced while (1) { ... k_sleep(1000); ...} with 
`k_timer_start(&smp_svr_timer, K_MSEC(1000), 0);`

And even after that things are working as expected.